tracer/src/Datadog.Trace/Sampling/DefaultSamplingRule.cs

```diff
@@ -34,6 +36,7 @@ public float GetSamplingRate(Span span)
 
             if (_sampleRates.Count == 0)
             {
+                span.SetMetric(Metrics.SamplingAgentDecision, 1); // Keep it to ease investigations
```

Comment:
We should confirm if this is correct behavior. When we send the first payload to the Agent, we use the default sampling rate (100% == set everything to `AUTO_KEEP`) because we don't have any rates from the Agent yet. It seems weird to set a tag that says "we used this sampling rate from the agent" when it wasn't from the agent.

The same applies even when count > 0, but we didn't find a matching sampling rate (so we fallback to 1).

Maybe the only change we need in this PR is if we get a 1 from the Agent, then don't skip it as an optimization.

See also: https://github.com/DataDog/dd-trace-dotnet/blob/4e8c080fe98c3c7c5d0409dd7a9407f005695bdf/tracer/src/Datadog.Trace/Metrics.cs#L12-L16

Reply:
> It seems weird to set a tag that says "we used this sampling rate from the agent" when it wasn't actually from the agent.

Checked with Raphael, he told me they never thought of that case and he told me we should add it. I'll ask Brett to make sure that's what we do in other tracers

> The same applies even when count > 0, but we didn't find a matching sampling rate (so we fallback to 1).

That case shouldn't happen anymore, cause if we don't have a rule, we use the default one sent by the agent.

Reply:
Thanks for confirming!